docs/src/main/asciidoc/building-my-first-extension-guide.adoc

```diff
@@ -0,0 +1,656 @@
+
+[source, xml]
+-------------
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>io.quarkus</groupId>
+        <artifactId>quarkus-sample-greeting-parent</artifactId>
+        <version>999-SNAPSHOT</version>
+        <relativePath>../pom.xml</relativePath>
+    </parent>
+
+    <artifactId>quarkus-sample-greeting</artifactId> <!--1-->
+    <name>Quarkus - Sample Greeting Extension - Runtime</name>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>io.quarkus</groupId>
+                <artifactId>quarkus-bootstrap-maven-plugin</artifactId> <!--2-->
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
+                    <annotationProcessorPaths>
+                        <path>
+                            <groupId>io.quarkus</groupId>
+                            <artifactId>quarkus-extension-processor</artifactId>  <!--3-->
+                            <version>${project.version}</version>
+                        </path>
+                    </annotationProcessorPaths>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+</project>
+-------------
+
+The key points are:
+
+<1> By convention, the runtime module has no suffix (`sample-greeting`) as it is the artifact exposed to the end user.
+<2> We add  the `quarkus-bootstrap-maven-plugin` to generate the Quarkus extension descriptor included into the runtime artifact which links it with the corresponding deployment artifact.
+<3> We add  the `quarkus-extension-processor` to the compiler annotation processors.
+
+=== Extension publication
+
+*For internal Quarkus usage*
+The `create-extension` command automatically added your extension modules in the appropriate descriptors:
+
+* Your extension (`sample-greeting`) is declared as a sub-module of `quarkus-extensions-parent` (`quarkus/extensions/pom.xml`)
+* The deployment module (`quarkus-sample-greeting-deployment`) is declared as a dependency in the `quarkus-bom-deployment` (`quarkus/bom/deployment/pom.xml) thus can be used by other projects.
+* The runtime module (`quarkus-sample-greeting`) is declared as a dependency in the `quarkus-bom-runtime` (`quarkus/bom/runtime/pom.xml) thus can be used by other projects.
+
+*For application developers*
+At this point, your extension is initialized, to make it visible when the developers list all Quarkus's extensions (ie. `mvn quarkus:list-extensions`) you should update the `quarkus/devtools/common/src/main/filtered/extensions.json` file.
+
+[source, json]
```

Comment:
We have moved to a different model. Now the file is this
https://github.com/quarkusio/quarkus/blob/master/extensions/hibernate-orm/runtime/src/main/resources/META-INF/quarkus-extension.yaml

@aloubyansky can you guide Daniel through the changes we made here.

Reply:
`quarkus/devtools/common/src/main/filtered/extensions.json` is actually gone. This should be removed. To be listed the extensions has to be added to either `quarkus-bom` (if it really belongs to the Quarkus repo, which I think should be an exceptional case now) or `quarkus-universe-bom` https://github.com/quarkusio/quarkus-platform/blob/master/bom/runtime/pom.xml (and the corresponding `-bom-deployment`).

The thing is that we are not going to be accepting arbitrary extensions in Quarkus repo any more, so we should not be promoting development of extensions in Quarkus repo.

Registering an extension in the universe is a second step. The extension has to be released first and be available in a Maven repo.